Makefile.sub

```diff
@@ -32,21 +32,26 @@ BLUE=\033[1;34m
 hls-test-run: hls-test
 
 .PHONY: hls-test
-hls-test: hls-test-base hls-test-dynamatic hls-test-benchmark hls-test-polybench
+hls-test: \
+	hls-test-base \
+	hls-test-benchmark \
+	hls-test-polybench \
+	hls-test-dynamatic \
 
 # Generic build target used by the various tests
+.PRECIOUS: $(HLS_TEST_BUILD)/%.hls
 $(HLS_TEST_BUILD)/%.hls: $(HLS_TEST_SRC)/%.c
 	@mkdir -p $(@D)
 	@set -e && printf '$(BLUE)Building: $(NC)%s\n' $*
-	@$(JHLS) $^ $(HLS_TEST_ADDITIONAL_SRC) $(HLS_TEST_ADDITIONAL_FLAGS) --circt --hls-function=$(HLS_TEST_FUNCTION) -o $@ > /dev/null
+	@$(JHLS) $^ $(HLS_TEST_ADDITIONAL_SRC) $(HLS_TEST_ADDITIONAL_FLAGS) --circt --hls-function=kernel -o $@ > /dev/null
 	@for file in $@.re*.ll ; do $(LLC) -O0 --relocation-model=pic -filetype=obj -o $$file.o $$file ; done
-	@TMPDIR=`mktemp -d`; \
+	@+TMPDIR=`mktemp -d`; \
 	$(FIRTOOL) -format=fir --verilog $@.fir > $$TMPDIR/jlm_hls.v; \
-	VERILATOR_ROOT=/usr/share/verilator verilator_bin $(VERILATOR_TRACE) --cc --build --exe -Wno-WIDTH -j -Mdir $$TMPDIR -MAKEFLAGS CXX=g++ -CFLAGS -g --assert -CFLAGS " -fPIC" -o $@ $$TMPDIR/jlm_hls.v $@*.o $@.harness.cpp > /dev/null
+	VERILATOR_ROOT=/usr/share/verilator verilator_bin $(VERILATOR_TRACE) --cc --build --exe -Wno-WIDTH -Mdir $$TMPDIR -MAKEFLAGS CXX=g++ -CFLAGS -g --assert -CFLAGS " -fPIC" -o $@ $$TMPDIR/jlm_hls.v $@*.o $@.harness.cpp > /dev/null
```

Comment:
So If I understand this correctly, then this change will start several instances of verilator instead of handing the `-j` option over to verilator and let it do the parallelization?

Reply:
Verilator also use make for building the simulation executable. This causes some issues as make is asked to run in parallel multiple times, creating more threads than originally asked for. Since the verilator step is just one out of many: running jhls, compiling llvm-ir to object files, converting firrtl to verilog, and running the benchmarks, there is enough work to do in parallel to run verilator single threaded.